### ISSUE TITLE
TABLESAMPLE SYSTEM prevents scan/filter/project operator fusion

### ISSUE THREAD

**Issue by dain** [prestosql/presto#29]
The `TABLESAMPLE` node in the plan for the following query prevents operator fusion:
```
presto> explain (type distributed) select length(name) from tpch.tiny.nation TABLESAMPLE system (50);
                                      Query Plan                                       
---------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]                                                                   
     Output layout: [length]                                                           
     Output partitioning: SINGLE []                                                    
     Grouped Execution: false                                                          
     - Output[_col0] => [length:bigint]                                                
             _col0 := length                                                           
         - RemoteSource[1] => [length:bigint]                                          
                                                                                       
 Fragment 1 [SOURCE]                                                                   
     Output layout: [length]                                                           
     Output partitioning: SINGLE []                                                    
     Grouped Execution: false                                                          
     - Project[] => [length:bigint]                                                    
             length := "length"("name")                                                
         - Sample[SYSTEM: 0.5] => [name:varchar(25)]                                   
             - TableScan[tpch:nation:sf0.01, grouped = false] => [name:varchar(25)]    
                     Cost: {rows: 25 (302B), cpu: 302.00, memory: 0.00, network: 0.00} 
                     name := tpch:name                                                 
 ```

**Comment by dain** [comment]
@Praveen2112, I think that will work, but we should only do this if `sourceNode.getSampleType() == SampleNode.Type.SYSTEM`.  I'd like to hear from someone more involved in the optimizer (maybe @martint or @findepi), if we should instead be optimizing away this node instead of handling it in `LocalExecutionPlanner`

**Comment by findepi** [comment]
@dain i prefer not to comment. I never used sampling yet.

**Comment by martint** [comment]
We need to add the ability to push down sampling into connectors (as part of picking table layout) and let the optimizer remove it from the plan if it's supported by the connector.

**Comment by dain** [comment]
I suggest we go ahead with this simple workaround for now.

**Comment by dain** [comment]
Fixed by #78
